### PR TITLE
openTab - incognito window

### PR DESCRIPTION
**Pull request by saikrishna321**
Fixes #1572 #1282 

Signed-off-by: saikrishna321 <saikrishna321@yahoo.com>

### CODE REVIEW

**Comment by zabil** [comment]
I did one round of testing with the following. 

```
const { openBrowser, goto, openTab, openIncognitoWindow, closeTab, closeIncognitoWindow, closeBrowser } = require('taiko');
(async () => {
    try {
        await openBrowser();
        await goto('google.com');
        await openTab('github.com');
        await openIncognitoWindow({name: 'gauge'});
        await goto('gauge.org');
        await openTab('taiko.dev');
        await closeTab();
        await closeIncognitoWindow('gauge');
        await closeTab();
    } catch (error) {
        console.error(error);
    } finally {
        await closeBrowser();
    }
})();
```

It looks good so far, will try more combinations.

**Comment by saikrishna321** [comment]
@zabil Thanks for looking at it! `await openIncognitoWindow({name: 'gauge'})` is an Incognito window with a tab name `gauge` so when users says `switchTo({name: 'gauge')` it switches to tab and not to the window. With that `await closeIncognitoWindow('gauge');` will close the tab {name: 'gauge'} and not the entire window. Looking into how to fix this. @NivedhaSenthil pointed me to use different API. 


Try scenario's by opening two IncognitoWindow and play around with openTab and switchTo

**Review by zabil — CHANGES_REQUESTED** [review]
Script

```
const { openBrowser, goto, openTab, openIncognitoWindow, switchTo, closeTab, closeBrowser } = require('taiko');
(async () => {
    try {
        await openBrowser();
        await goto('google.com');
        await openTab('github.com', { name: 'github'});
        await openIncognitoWindow({name: 'gauge'});
        await openTab('github.com', { name: 'github2'});
        await switchTo(/github/);
        await switchTo({name: 'github'});
        await closeTab();
    } catch (error) {
        console.error(error);
    } finally {
        await closeBrowser();
    }
})();
```

The `closeTab` at the end of the script close the last tab on the incognito window instead of the tab that it switched to on the main window.

File: lib/taiko.js [hunk 1]
```diff
@@ -622,8 +622,16 @@ module.exports.openTab = async (
     await connect_to_cri(target);
     targetHandler.register(options.name, target.id);
   };
+  if (this.currentContext) {
+    const browserContextId = this.currentContext.browserContextId;
+    const { targetId } = await _client.Target.createTarget({ url: targetUrl, browserContextId });
+    targetHandler.register(options.name, targetId);
+    await connect_to_cri(targetId, options);
+    await dom.getDocument();
```
Comment (NivedhaSenthil) [inline lib/taiko.js:630]: we dont need`dom.getDocument()` anymore as we use objectID instead of nodeID

**Review by NivedhaSenthil — CHANGES_REQUESTED** [review]

**Comment by saikrishna321** [comment]
@zabil @NivedhaSenthil `closeTab` API currently closed all the matching tabs with given identifier. Should we only close the current active tab. 

For example: 

`
await openBrowser();
        await goto('google.com');
        await openTab('github.com', { name: 'github'});
        await openTab('github.com', { name: 'github1'});
        await closeTab(); // This will close both the github.com tabs. I'm expecting the tab with name github1 should only be closed.
`

**Comment by NivedhaSenthil** [comment]
I feel the current behavior is correct.. if user wants to close the active tab they can simply say `closeTab()` when an URL is explicitly passed I would expect Taiko to close all the tabs matching the URL

**Comment by NivedhaSenthil** [comment]
Also corresponding to above comment.. I would expect the below code to close only the active tab and not both the tabs matching `gauge.org` ...its closing all the matching tabs with the changes in PR

```
(async () => {
    try {
        await openBrowser();
        await goto("google.com");
        await openTab("gauge.org");
        await openTab("gauge.org");
        await closeTab();
    } catch (error) {
        console.error(error);
    } finally {
        await closeBrowser();
    }
})();
```

**Comment by NivedhaSenthil** [comment]
closing in favor of #1617

**Pull request closed without merging**